doc/devel/MEP/MEP28.rst

```diff
@@ -0,0 +1,177 @@
+
+  "AB"
+  ┌────────┐┌────────┐
+  │ A      ││ B      │
+  │        ││        │
+  │        ││        │
+  └────────┘└────────┘
+
+  "ABB"
+  ┌──────┐┌──────────┐
+  │ A    ││ B        │
+  │      ││          │
+  │      ││          │
+  └──────┘└──────────┘
+
+  "ABD"
+  "CCD"
+  ┌───────┐┌───────┐┌───────┐
+  │ A     ││ B     ││ D     │
+  │       ││       ││       │
+  │       ││       ││       │
+  └───────┘└───────┘│       │
+  ┌────────────────┐│       │
+  │ C              ││       │
+  │                ││       │
+  │                ││       │
+  └────────────────┘└───────┘
+
+  "AaBb"
+  ┌───────┐┌─┐┌───────┐┌─┐
+  │ A     ││ ││ B     ││ │
+  │       ││ ││       ││ │
+  │       ││ ││       ││ │
+  └───────┘└─┘└───────┘└─┘
+
+  "  b  "
+  "aABCc"
+              ┌───────┐
+              └───────┘
+  ┌─┐┌───────┐┌───────┐┌───────┐┌─┐
+  │ ││ A     ││ B     ││ C     ││ │
+  │ ││       ││       ││       ││ │
+  │ ││       ││       ││       ││ │
+  └─┘└───────┘└───────┘└───────┘└─┘
+
+
+Syntax
+------
+
+A subplot is described using a capital letter, from ``A`` to ``Z`` such that
+only 26 subplots can be specified at once. Colorbars are specified using
+non-capital letters, from ``a`` to ``z``. The main command is the ``layout``
+command that returns a list of axes (subplots and colorbars) following the
+lexicographic order, mixing capital and small letters. For example::
+
+  A,B,C = layout("ABC")
+  A,B,b = layout("ABb")
+
+A layout can span several lines. In such a case, it is necessary the specify the
+layout using an array of strings::
+
+  A,B = layout(["A",
+                "B"])
+
+The size of a subplot is relative to the size of its string representation, a
+single letter representing the logical unit. For example::
+
+
+  A,B = layout("AAB")
+
+means ``A`` is two logical block wide while ``B`` is one logical block
+wide. The final result should be ``A`` width to be 2/3 while ``B`` width should
+be 1/3.
+
+Colorbars can be specified using a small letter and has either a fixed width
+(vertical colorbar) or a fixed height (horizontal colorbar). For example::
+
+  A,a = layout("Aa")
+
+The exact width or height of colorbars must be specified throught matplotlib
+properties. The difficulty being the attachment of the colorbar to the right
+figure. In the example above, the size of ``A`` should be logical block and the
+colorbar ``a`` should be attached to the right side of ``A``.
+
+  A,a,B = layout("AaB")
+
+In such a case, ``A`` and ``B`` size is one logical block and ``a`` should be
+attached to the right side of ``A``.
+
+
+
+
+Subplot aspects
+---------------
+
+It is not yet clear how subplot aspects can be specified and how this
+would constrain the overall layout. We need more examples.
+
+
+Implementation
+==============
+
+
+
+**Note**: Implementation probably requires a geometry manager able to enforce
+constraints but it is not yet clear if a full geometry manager is required
+(constraints might be relatively easy to solve).
```

Comment:
As pointed out by @Tillsten, gridspec may meet the requirements, or may given some enhancements.  If not, maybe a list of the reasons why it wouldn't work would be valuable here.